index.js

```diff
@@ -0,0 +1,222 @@
+    this.transformTop = 0;
+
+    requestAnimationFrame(() => this.el.removeAttribute('style'));
+  }
+
+  /**
+   * Resize handler
+   *
+   */
+  onResize() {
+    if (this.matchesMedia()) {
+      this.calcBounds();
+    } else {
+      this.clearTransfoms();
+      this.destroyListeners();
+
+      if (this.observer) {
+        this.observer.disconnect();
+      }
+    }
+  }
+
+  /**
+   * Determines whether the matchMedia settings matches the viewport size or not
+   *
+   * @returns {boolean}
+   */
+  matchesMedia() {
+    return this.options.matchMedia.matches;
+  }
+
+  /**
+   * Scroll handler
+   *
+   */
+  onScroll() {
+    const scrollTop = this.scrollTop();
+
+    if (scrollTop < this.startPosition || scrollTop > this.endPosition) {
+      return;
+    }
+
+    const scroll = scrollTop - this.startPosition;
+
+    requestAnimationFrame(() => {
+      this.transformTop = scroll;
+      this.el.style.transform = `translate3d(0, ${scroll}px, 0)`;
+    });
+
+    if (!this.targets.length) {
+      return;
+    }
+
+    const [{ el: targetEl }] = this.targets.filter(target => scrollTop >= target.top).slice(-1);
+
+    if (targetEl && targetEl !== this.current) {
+      fire(this.el, Events.Leave, { detail: this.current });
+      fire(this.el, Events.Enter, { detail: targetEl });
+
+      this.current = targetEl;
+
+      this.inView({ el: this.el, target: this.current });
+    }
+  }
+
+  /**
+   * On intersect handler
+   *
+   * @param {array} entries Ofserved items
+   */
+  onIntersect(entries) {
+    entries.forEach(entry => {
+      if (entry.isIntersecting) {
+        this.calcBounds();
+        this.addListeners();
+      } else {
+        this.destroyListeners();
+      }
+    });
+  }
+
+  /**
+   * Called upon element it's visible
+   *
+   * @param {HTMLElement} element Current target element
+   */
+  inView(element) {}
+
+  /**
+   * Calculate element boundaries
+   *
+   */
+  calcBounds() {
+    const scrollY = this.scrollTop();
+    const { top: startElTop, height: startElHeight } = this.el.getBoundingClientRect();
+    const { height: endElHeight } = this.options.container.getBoundingClientRect();
+
+    this.startPosition = startElTop + scrollY - this.transformTop + this.options.offset;
+    this.endPosition = this.startPosition + (endElHeight - startElHeight);
+
+    this.targets = this.targets.map(({ el }) => {
+      const { top } = el.getBoundingClientRect();
+
+      return {
+        el,
+        top: scrollY + top - (startElHeight / 2)
+      };
+    });
+  }
+
+  /**
+   * Returns the number of pixels the page is currently vertically scrolled
+   *
+   * @returns {number}
+   */
+  scrollTop() {
+    return window.pageYOffset || document.documentElement.scrollTop;
+  }
+}
+
+const init = () => {
+  const elements = find('[data-sticky]');
+  const instances = [];
+
+  elements.forEach(element => instances.push(new Sticky(element)));
+
+  on('img[data-scroll]', 'load', () => fire(window, 'resize'));
```

Comment:
I think this is coupling this to `scroll-triggers` which we shouldn't do IMHO. We should probably do a PR on ScrollTriggers anyways to handle this and also to the lazy load library.